### PR TITLE
fix(generator): preserve gcloud 585 command coverage

### LINKED ISSUE #5154: Core concurrency tests time out before blocking fixture workers start
## Failure

The Ubuntu pipeline for #5152 failed three core concurrency tests at head 9026f0e948e653ce4a5caaf14a21e47fba19f3e8:
https://github.com/thomhurst/ModularPipelines/actions/runs/35029946342/job/104588171167

- Dispose_WhileStateTransitionIsInFlight_DoesNotThrow: transitionEntered did not arrive within five seconds (ModuleSchedulerDisposalTests.cs:41).
- DirectConsoleWrite_DefersRegistrationUntilWriteCompletes: writeStarted.Wait returned false (SecretMaskingPatternTests.cs:1348).
- StableSecretEmission_UsesEvenDeferredSnapshotVersions: emissionStarted.Wait returned false (SecretMaskingPatternTests.cs:1576).

The latter two tests synchronously wait for workers queued with Task.Run. All three intentionally block worker callbacks on explicit release gates, making startup depend on shared thread-pool availability under parallel CI load. Other tests in SecretMaskingPatternTests already use dedicated LongRunning tasks for this pattern. The generator tests passed; no rerun has been requested.

## Expected outcome

- Reproduce the scheduling dependency under a constrained thread pool where practical.
- Run intentionally blocking fixture workers independently of the shared pool, preserving the actual overlap, secret registration, snapshot-version, and disposal assertions.
- Release and observe held work even if startup fails.
- Keep existing timeout values; pass focused core tests and Linux CI.

Related timing investigation: #5101 concerns a separate distributed-cache test.

### PR DESCRIPTION
**Pull request by thomhurst**
Google Cloud SDK 585.0.0 help caused the generator to drop commands when wrapped option values looked like operands, nested option-only groups looked ambiguous, or inherited group prose looked like collection syntax. This fixes those cases in the scraper/shared parser and replays all 54 affected SDK fixtures. All previously failing fixtures now retain their commands. The handwritten ParseUsageSynopsis API change is tracked.

This also fixes the three core concurrency-test failures seen in this PR's Ubuntu pipeline (issue #5154). Intentionally blocking fixture workers now use dedicated LongRunning tasks. The disposal test releases/drains work even when startup fails and verifies the transition remains in flight. Existing timeout values and secret-registration/snapshot assertions remain intact.

Validation:
- 361 focused AWS/gcloud/shared-parser tests pass after rebasing onto main, including the 54 SDK fixtures.
- A one-worker thread pool reproduces both logging failures before the synchronization fix; all three affected tests pass afterward in 478 ms. No thread-pool setting is committed.
- All 89 related core disposal/secret-masking tests pass with normal runtime settings.
- Generator and core-test scoped severity-info formatting pass. Full Linux CI is rerunning with the fixes.

#5147 and #5150 are merged. Merge this source fix before fresh gcloud generation from main. Issue #5146 stays open until that complete generation succeeds; no generated options are patched here.

Closes #5154.

Refs #5146.

## Summary by CodeRabbit

* **New Features**
  * Expanded Google Cloud CLI coverage for version 585.0.0 across Compute, Dataproc, IAM, Storage, Transfer, Spanner, configuration, and additional services.
  * Improved recognition of wrapped option values, positional arguments, boolean options, repeatable parameters, and complex command syntax.

* **Tests**
  * Added broader validation using captured Google Cloud command help and edge-case argument formats.
  * Improved test reliability for operations involving blocked or long-running workers.

### CODE REVIEW

File: tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs [hunk 1]
```diff
@@ -1332,7 +1330,7 @@ private static bool TryParseNestedOperandGroup(
         }
 
         if ((!content.Contains('[') && !nestedTokens.Any(nestedToken => GetOptionSwitches(nestedToken).Count > 0))
-            || content.Contains('|'))
+            || SplitTopLevelAlternatives(content).Count > 1)
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs:1340]: Checked this against remote head 8cc5283a083f0d43e5cb19fda0f52beb9c365168 after rebasing onto the latest #5147 fixes. All 13 existing Does_Not_Model_Option_Value_Alternatives_As_Operands cases pass, including the exact detached angle-bracket, brace, and square-bracket examples; all 270 focused parser/resource tests pass. ParseOperandTokens checks TryHandleOptionToken and IsNonOperandSyntax before TryParseNestedOperandGroup, so these option-value tokens do not take the proposed fallback path. No additional early exit is needed for these cases.

**Comment by thomhurst** [comment]
Addresses the two non-blocking notes in https://github.com/thomhurst/ModularPipelines/pull/5152#issuecomment-5688611665:

1. ResourceOperandPattern requires an uppercase operand name beginning with A-Z, optionally after a compound prefix. That excludes empty names and standalone option switches before GetOperandPropertyName is called. The captured compound-operand regressions cover this invariant. I kept the canonical property name instead of falling back to raw help text, which could introduce an invalid or inconsistent identifier.
2. Trusting the option-aware reparse is intentional. Commit 18444a76d8d0489884f5bfb5fc577b5836dc49f7 on #5147 fixes the concrete `[--verbose RESOURCE]` case: the coarse parse associates RESOURCE with an unknown option, while resolved flag metadata proves it is a positional operand. Intersecting both parses discarded the required conditional constraint. The new regression plus 270 parser/resource tests pass.

Current remote head f8da14a248026843df6146714b854a26286df70f includes the parent fixes and API tracking for the newly overridable ParseUsageSynopsis method. The generator solution Release build passes with zero warnings and errors.

**Comment by thomhurst** [comment]
Addresses the clarity notes in https://github.com/thomhurst/ModularPipelines/pull/5152#issuecomment-5689032288. Commit 724659ea66 adds a clear assertion before slicing fixture NAME lines and documents why both shape checks use option-local prose: inherited group descriptions can describe sibling values. All 54 SDK fixtures and scoped formatting pass. The potential shared-parser generalization and regex memoization remain future work without a demonstrated need in this scope.

These changes are included in verified remote head 363bc39cbe56f4a7d01449c8727864f8efb9eac0, rebased onto main after #5147 merged. That head also fixes the separate core test scheduling failures from the latest Ubuntu run; details and reproduction are tracked in #5154. All 361 focused generator tests and 89 related core tests pass locally.

**Comment by thomhurst** [comment]
The generator changes implement #5146: preserve all commands rejected by the SDK 585 scrape, including structured and multiword option values and option-local shape validation. That issue is referenced in the PR description and intentionally remains open until a fresh full generation verifies command coverage after this source fix reaches main. #5154 separately tracks the core test failures exposed by this PR's CI; commit 363bc39cbe fixes those reproduced failures.

For the docstring warning, the repository does not require XML comments on private helpers or test methods. The new private ValidateRepeatableOption helper in 8d24e48e3ae57d534fce5b3c0c8f2445658fb577 names the validation it performs, and the caller documents the non-obvious option-local description rule. Adding boilerplate solely to meet the bot's percentage would not clarify the implementation. All 361 focused generator tests and scoped formatting pass; CodeFactor now passes as well.
